File: app/src/main/java/com/garymcgowan/moviepedia/view/BasePresenter.java
package com.garymcgowan.moviepedia.view;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

/**
 * Created by garymcgowan on 13/05/2017.
 */

public class BasePresenter {

    private CompositeDisposable disposables = new CompositeDisposable();


    public void addDisposable(Disposable disposable) {
        disposables.add(disposable);
    }

    public void clearDisposables() {
        disposables.clear();
    }
}

File: app/src/main/java/com/garymcgowan/moviepedia/view/BasePresenter.kt
package com.garymcgowan.moviepedia.view

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable
import java.lang.ref.WeakReference

abstract class BasePresenter<V> {

    private var view: WeakReference<V>? = null

    private val disposables = CompositeDisposable()

    fun takeView(view: V) {
        this.view = WeakReference(view)
    }
    fun dropView() {
        this.view = null
        this.disposables.clear()
    }

    fun getView() = this.view?.get()

    fun addDisposable(disposable: Disposable) {
        disposables.add(disposable)
    }

    fun clearDisposables() {
        disposables.clear()
    }
}
